### ISSUE TITLE
Implement new access control

### ISSUE THREAD

**Issue by rexxars** [imbo/imboclient-php#102]
With the upcoming access control changes, we'll have to modify the client a bit.

1. There needs to be a clear distinction between a `user` and a `public key`
2. There should possibly be a way to specify public and private keys on a per-command basis
3. Public key should be sent as header instead of assuming it's present in the URL

Obviously, documentation also needs to be updated to reflect the changes.

Does anyone have any suggestions on how the API would look best in regards to point 2?

**Comment by kbrabrand** [comment]
With regards to point 2 I think adding an optional parameter to the methods would be OK. Most of the methods have less than three parameters, and in most cases there will be no need for the public/private keys.

Choosing this approach would mean that the client is instantiated in the same way as before and the change will be non-breaking. If the user [of the client] wants to specify an alternative key pair, it can be done using the options array which holds three optional properties.

In order to fetch images for `user1` and `user2` you would use the client in the following way;

```php
$options = [
    'publicKey' => 'my-alternative-publickey',
    'privateKey' => 'secret',
    'users' => [
        'user1',
        'user2'
    ]
];

$client->getImages($query, $options);
```

**Comment by rexxars** [comment]
We decided to go with creating new clients of altering each and every method to support options. Implemented in #104.
